### PR TITLE
Fix OAuth2

### PR DESCRIPTION
**Pull request by rmlopes**
Fix OAuth2. As per the example `BearerAuth` extends `BasicAuth`, as such rather than using `app.auth` class type, we first check for sentinel config vars than proceed with the remaining auth types. HTTPS scheme must be used with OAuth2 as such I also re-added `_get_scheme` which was removed in the last merges.

### CODE REVIEW

File: eve_swagger/objects.py [hunk 1]
```diff
@@ -16,6 +16,11 @@
 from .definitions import INFO, HOST
 
 
+def _get_scheme():
+    print(app.auth)
```
Comment (nicolaiarocci) [inline eve_swagger/objects.py]: could you please remove this line

Comment (rmlopes) [reply eve_swagger/objects.py]: It's done, and fixed the host url. Everything shoudl be fine now.

**Comment by rmlopes** [comment]
Thanks @DavidZisky, I was postponing black because it is only available for Python 3.6+ and I am running debian that is shipped with 3.5 and it is a bit delayed, even in the backports, but I have surrendered :). The build still fails, but the tests run clean on my system with 3.7.

**Comment by nicolaiarocci** [comment]
rebased and merged. thanks.

File: eve_swagger/objects.py [hunk 2]
```diff
@@ -16,6 +17,10 @@
 from .definitions import INFO, HOST
 
 
+def _get_scheme():
+    return "http" if app.auth is None else "https"
```
Comment (JoveYu) [inline eve_swagger/objects.py]: if use auth, then server is https?

**Comment by JoveYu** [comment]
the generator don't know what is the real server url, maybe you deploy nginx with ssl cert before eve

you can use SWAGGER_HOST to set real url if you use https, instead of change default

**Comment by nicolaiarocci** [comment]
feel free to submit a PR

**Comment by JoveYu** [comment]
i don't hava a project use oauth2, maybe @rmlopes can help to fix this

**Pull request closed without merging**